pkg/services/baremetal/remediation/remediation.go

```diff
@@ -59,9 +59,7 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
 				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
 				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-				return res, err
```

Comment:
I actually think that here we should still return an error even, as this error that we couldn't update the Machine object is unexpected